test/test_sync.cpp

```diff
@@ -275,14 +275,21 @@ TEST(SyncEvent, WaitFor)
         const steady_clock::time_point start = steady_clock::now();
         const bool on_timeout = !cond.wait_for(lock, timeout);
         const steady_clock::time_point stop = steady_clock::now();
+        const steady_clock::duration waittime = stop - start;
 #if defined(ENABLE_STDCXX_SYNC) || !defined(_WIN32)
         // This check somehow fails on AppVeyor Windows VM with VS 2015 and pthreads.
         // - SyncEvent::wait_for( 50us) took 6us
         // - SyncEvent::wait_for(100us) took 4us
         if (on_timeout) {
-            EXPECT_GE(count_microseconds(stop - start), tout_us);
+            EXPECT_GE(waittime, timeout);
```

Comment:
I suppose `tout_us` was used here instead of `duration` because GTest was not formatting the duration properly.
An error message can be added then.
```suggestion
            EXPECT_GE(waittime, timeout); << Expected << tout_us << waited << count_microseconds(waittime);
```

Reply:
The problem with this condition is that even this condition sometimes fails.

Reply:
I haven't seen any fail here outside the conditionals. Is that too little tolerance, or there's no guarantee at all?